### ISSUE TITLE
[BUG] einsum is not working

### ISSUE THREAD

**Issue by ZJUGuoShuai** [NVIDIA/MatX#167]
**Describe the bug**
`matx::cutensor::einsum` outputs wrong results.

**To Reproduce**
1. Create a 2x3 tensor `a`:
```cpp
auto a = matx::make_tensor<float, 2>({2, 3});
a.SetVals({
    {1, 2, 3},
    {4, 5, 6}
});
```
2. Perform einsum operation to reduce sum `a`:
```cpp
auto a_reduced = matx::make_tensor<float, 1>({3});
matx::cutensor::einsum(a_reduced, "ij->j", 0, a);
```
3. Print the result:
```cpp
cudaStreamSynchronize(0);
a_reduced.Print();
```
![image](https://user-images.githubusercontent.com/22856433/164877917-ccf9e706-280d-495b-93c3-2a247ad35292.png)

The shape is correct but all values are zeros.

**Expected behavior**

NumPy result:
![image](https://user-images.githubusercontent.com/22856433/164877937-fee3df69-121e-4704-a9e3-7426d052eb9c.png)

**System details (please complete the following information):**
 - OS: Ubuntu 20.04
 - CUDA version: CUDA 11.4
 - g++ version: 9.3.0

**Comment by cliffburdick** [comment]
thanks for reporting this. we don't currently have a reduction test using einsum, so we'll add one in. if you're just looking for the same functionality you can do it by using sum() with a 2D input and 1D output.

**Comment by ZJUGuoShuai** [comment]
@cliffburdick I do checked `sum()`, but it can only reduce over the innermost dimensions. Using `einsum()` I can sum over arbitrary dimensions, like PyTorch `torch.sum()` with `dim=k`.

**Comment by cliffburdick** [comment]
have you tried calling Permute() before sum or after you looking for something else?

**Comment by ZJUGuoShuai** [comment]
@cliffburdick `Permute()` before `sum()` works fine, thank you.

**Comment by cliffburdick** [comment]
Hi @ZJUGuoShuai , it turns out cuTENSOR doesn't support axis summing yet. We'll update the code to dispatch the regular sum routine if it detects that in the einsum pattern.

**Comment by cliffburdick** [comment]
@ZJUGuoShuai it turns out that reductions are supported in the new cutensor we just updated to. I'll debug why it's not working, but you should get a significant speed up over the permute method once it's working.

**Comment by cliffburdick** [comment]
@ZJUGuoShuai I took your exact code and ran it in the version I pushed 2 days ago, and it appears to work:

```
 auto a = matx::make_tensor<float, 2>({2, 3});
  a.SetVals({
      {1, 2, 3},
      {4, 5, 6}
  });  

  auto a_reduced = matx::make_tensor<float, 1>({3});
  matx::cutensor::einsum(a_reduced, "ij->j", 0, a);
    
  a_reduced.Print();

000000: 5.0000e+00
000001: 7.0000e+00
000002: 9.0000e+00
```

**Comment by ZJUGuoShuai** [comment]
@cliffburdick I updated MatX to the lastest commit, and the problem disappeared. I'm closing this issue, thanks a lot!

**Comment by cliffburdick** [comment]
Hi @ZJUGuoShuai I added a unit test using your example to show the syntax as well. I would recommend using this over `sum()` since it should be much faster:

https://github.com/NVIDIA/MatX/blob/main/test/00_tensor/EinsumTests.cu#L210
